### ISSUE TITLE
"Chats failed to load"

### ISSUE THREAD

**Issue by AwareDescription25** [irsat000/CAI-Tools#90]
When fetching chat histories to download, almost every time I get the message "X amount of chats failed to load." Why is this? It would be helpful to at least get a list of which ones failed to load so I can manually download them. But it's really frustrating when I'm trying to save a bot with 100+ chats and I can't see which ones it saved or not.
![Screenshot (1609)](https://github.com/user-attachments/assets/f18f6592-9b53-402a-ba4b-945e44201e0c)

**Comment by irsat000** [comment]
It normally shouldn't even fail one chat. That's why we don't need a list of failed chats. I will check what is wrong.

Are other chats downloaded?

**Comment by AwareDescription25** [comment]
> It normally shouldn't even fail one chat. That's why we don't need a list of failed chats. I will check what is wrong.
> 
> Are other chats downloaded?

Yeah, it does download a good portion of them. It seems to be the same ones every time. (for example I have 161 chats with this bot, but it's the same 42 that always fail to fetch)

**Comment by irsat000** [comment]
I have no idea what the problem is but I made a test version that will download the list of errors as text file. Perhaps that can help me figure it out.

[CAI-Tools-HistoryFixTest-1.zip](https://github.com/user-attachments/files/17982262/CAI-Tools-HistoryFixTest-1.zip)

How to install: https://www.youtube.com/watch?v=oswjtLwCUqg

**Comment by AwareDescription25** [comment]
> I have no idea what the problem is but I made a test version that will download the list of errors as text file. Perhaps that can help me figure it out.
> 
> [CAI-Tools-HistoryFixTest-1.zip](https://github.com/user-attachments/files/17982262/CAI-Tools-HistoryFixTest-1.zip)
> 
> How to install: https://www.youtube.com/watch?v=oswjtLwCUqg

Alright. I tested it. The text file output just shows "Couldn't fetch chat chunk - Status: 403" for every failed chat, with no other information.
![Screenshot (1642)](https://github.com/user-attachments/assets/907c0321-ea6e-4964-953f-c2669d2b7a41)

**Comment by irsat000** [comment]
Status 403 means "Forbidden". You can't get those chats for some reason. I can't fix this but I can give better errors than just saying "failed to load".

I don't even know what could possibly cause this. Since it's pretty unbelievable that you can't reach your own chats, it leads me to think that in those chats we somehow can't provide the access token and can't authenticate.

Thank you, I know where to check now. Meanwhile, you can brainstorm on your own, maybe those chats are flagged or something, never heard of such a thing though.

I can also come up with more test, for example give links next to the failed statuses so you could try seeing them normally.

**Comment by AwareDescription25** [comment]
> Thank you, I know where to check now. Meanwhile, you can brainstorm on your own, maybe those chats are flagged or something, never heard of such a thing though.

I just noticed something odd...
In one of my bots the extension detects 27 chats, when I only have 20 unarchived chats with that bot. It gave me "7 chats failed to load" yet still downloaded all 20 chats. Could it have something to do with the archived chats? Correct me if I'm wrong, but it looks like it's designed to skip over the empty chats that only have the first greeting message... I have 12 archived chats with that bot, and 5 of those are just the greeting, leaving 7 full archived chats. After digging around, this seems to be the case with all bots I've chatted with. So it does actually seem to be downloading all my open chats lol

**Comment by irsat000** [comment]
Yeah I skip over chats with one message. But "failed to load" shouldn't have happened.

**Comment by irsat000** [comment]
Hold on, archived chats?
